### ISSUE TITLE
Crash when Resetting Password

### ISSUE THREAD

**Issue by aarons2000** [jellyfin/jellyfin#1577]
I am unable to reset my password using the 'Forgor Password' function at login. The file saves successfuly at "/var/lib/jellyfin/passwordreset1.json" but when I try to enter the PIN to reset the password nothing happens. Looking at developer tools, I get the following errors:
![image](https://user-images.githubusercontent.com/39073142/62012885-be1e4e00-b183-11e9-87c9-6f9a5be6e82b.png)

I have tried this oon different browsers, but the problem seems to be that the page "http://HOSTNAME:8096/emby/Users/ForgotPassword/Pin" doesn't exist.

**Comment by onny** [comment]
Could you please tell which Jellyfin version your are using and how you installed it?

**Comment by anthonylavado** [comment]
@aarons2000 In addition to providing the information that @onny has asked for, would you be able to try this page?

`http://HOSTNAME:8096/web/index.html#!/forgotpasswordpin.html`

**Comment by aarons2000** [comment]
Sure,
Sorry I'm so late replying I've been away.

**Comment by aarons2000** [comment]
> 
> 
> Could you please tell which Jellyfin version your are using and how you installed it?
> 
> Best regards,
> Jonas

When I run `jellyfin --version`, I get an output of `Jellyfin.Server 10.3.7.0`, so I'm running version 10.3.7.0.
I installed it following your instructions for Ubuntu here https://jellyfin.readthedocs.io/en/latest/administrator-docs/installing/#ubuntu.



> 
> 
> @aarons2000 In addition to providing the information that @onny has asked for, would you be able to try this page?
> 
> `http://HOSTNAME:8096/web/index.html#!/forgotpasswordpin.html`

That is the page I am trying to use to reset the password and it is the one I get the error on.

**Comment by s0urcelab** [comment]
same here, when submit PIN, dev console gives a 404 error:
request URL is : http://localhost:8096/emby/Users/ForgotPassword/Pin

**Comment by sachk** [comment]
Same issue here, 10.3.7.0 on Debian 10.

**Comment by jfml** [comment]
I have the same problem also on 10.3.7, running on a Synology via Docker 17.0.50. 

I'm super desperate to get a fix for this since I can't log into my admin account, so I'd be happy to help with any testing.

**Comment by anthonylavado** [comment]
For all with password issues, there is a workaround you can use in the meanwhile.

If you have access to the server config files, you can set a field in `system.xml` to create a new Admin account.

At the end of the file, set this to false:

```
<IsStartupWizardCompleted>true</IsStartupWizardCompleted>
```

Then, try the web interface again. It will be at the first time setup. Make a new account with a new name, and it will be an Admin account. You can reset any password you need to on other accounts. Once you’re done, log out of this new account, and log in as the new one you created. You can then delete the new account.

**Comment by jfml** [comment]
Can confirm that the method described above by @anthonylavado worked for me. 

Also if your account is gone (like mine was) it might be that it has been disabled because you entered a wrong password too often. I was able to enable the account by going to

/config/users/[username]/policy.xml

and setting <IsDisabled>true</IsDisabled> to false.

**Comment by JustAMan** [comment]
Could you please updating to Jellyfin 10.4.0 and re-checking?

**Comment by texneus** [comment]
Jellyfin 10.3.7 on Docker under Debian Buster:

Just chiming in to say I just had the same experience and was locked out of my admin account tonight.  This has happened to me several times now since I started using Jellyfin way back when it first came out.  Just like every other time this has happened, it just randomly decides not to let me in.  I was just logged in two days ago, but tonight it won't let me in at all.  The logs do not indicate any activity on the admin account in the interim that I can see.  The logs also show I mistyped the password once (and only once) tonight, then it disabled the account even though it is set for 5 attempts.

The whole reset password thing has never worked correctly, at least for me.  It tells you to go to a location on your sever for a file containing a PIN, only the file isn't just a PIN and has other data as well.  I take out what I think is a PIN (because it's not what I expect a PIN to be), Jellyfin accepts it, tells me to use the PIN I just entered as my new password, only to then tell me the username or password are wrong!  I don't even know if what I have is a PIN or not because this isn't very clear (Jellyfin doesn't seem to be helpful here because it seems to accept any "PIN" I choose to enter).   If it says it created a PIN in the file, then it should be very clear what that PIN is.  Ideally it should be the only thing that file contains or perhaps include some supplemental instructions.  I don't think I'm alone in expecting a PIN to be true to it's namesake and be a number (like 123456).  What it says is a PIN looks like a 8 bit hexadecimal numbers separated by dashes and in quotations:  {"Pin":"A2-70-C2-45",a bunch of other strings all run together on one line}.  Are the quotes part of the pin?  Am I supposed to remove the dashes or leave them?  Is this supposed to represent a 4 letter ASCII code?  Is some/all/none of data that appears after the final quote part of the PIN?  I have yet to figure this out.  I have not seen the errors others are reporting here, other than Jellyfin's propensity to slam the door and lock it.  The current password recovery has either always been broke or is too ambiguous for me to figure out how to use it.

@JustAMan, I was going to upgrade to 10.4 tonight but then this happened.  Since I'm not sure why I get locked out in the first place or see these errors, I'm not sure how much help I can be.  I'll play with this over the weekend.

For others who are also stuck:  The work around suggested by @anthonylavado does work and it's quite painless.  However, (at least for me) placing it at the end of the config.xml file just resulted in the extra line just being deleted when Jellyfin was restarted.  You need to find the existing line in the config.xml file and change it.  Mine was very near the top of the file.

Like @jfml, upon regaining access my admin account was not listed in Jellyfin, but his trick didn't work for me (maybe this is a Docker thing but the [username] directories were random characters...I had to hunt for it but I'm sure I found it).  I ended up creating a new admin account (reused the same name and everything).  I expected Jellyfin would complain but it accepted it no problems.  In the process it created a new [username] directory and the old one was empty so I just deleted it manually.

**Comment by JustAMan** [comment]
The reason I ask to re-test on 10.4.0 is that if this bug is only present in 10.3.7 then we most likely won't be fixing it (because it might be fixed in our later releases).

As for password reset being  funky in docker... that is somewhat known, but we haven't found volunteers to fix that yet.

Usability issues (like "PIN" being misnamed and unclear instructions on how to get it from the system) should be submitted as a separate issue.

**Comment by anthonylavado** [comment]
>Usability issues (like "PIN" being misnamed and unclear instructions on how to get it from the system) should be submitted as a separate issue.

For whoever creates this, I think the reason the PIN file has more stuff in it was to make it "script" friendly (hence the JSON). We should probably just make it more human friendly, or get better instructions.

Perhaps it would be better to call it "Reset Code" instead of "PIN", since it includes dashes and letters, not just numbers.

**Comment by cvium** [comment]
I vaguely remember fixing some PIN reset stuff in web. Make some noise if this issue persists.

**Comment by zynexiz** [comment]
Just noticed the same issue here, there's no way to change a users password. Logging in with admin, I tried to change the password, but it still require the old password before I can change it (which is quite odd when you are the administrator, I don't know the users password). Clicking reset password seemed to clear it, but still can't enter a new one. When trying to enter one, it just trow an error "Invalid username or password. Please try again".

I also tried resetting it from the login page, found the pin in the jellyfin folder on the server. But the pin don't work, still complains that the password is incorrect.

**Comment by kklot** [comment]
> For all with password issues, there is a workaround you can use in the meanwhile.
> 
> If you have access to the server config files, you can set a field in `system.xml` to create a new Admin account.
> 
> At the end of the file, set this to false:
> 
> ```
> <IsStartupWizardCompleted>true</IsStartupWizardCompleted>
> ```
> 
> Then, try the web interface again. It will be at the first time setup. Make a new account with a new name, and it will be an Admin account. You can reset any password you need to on other accounts. Once you’re done, log out of this new account, and log in as the old one you fixed. You can then delete the new account.

This does not work for me, it resets only the first user on the list but not let it be an admin (I had my admin account hidden from login screen).

**Comment by YouthfulRicker** [comment]
> For all with password issues, there is a workaround you can use in the meanwhile.
> 
> If you have access to the server config files, you can set a field in `system.xml` to create a new Admin account.
> 
> At the end of the file, set this to false:
> 
> ```
> <IsStartupWizardCompleted>true</IsStartupWizardCompleted>
> ```
> 
> Then, try the web interface again. It will be at the first time setup. Make a new account with a new name, and it will be an Admin account. You can reset any password you need to on other accounts. Once you’re done, log out of this new account, and log in as the old one you fixed. You can then delete the new account.
> - @anthonylavado 

This worked for me, except the xml tag was actually on line 4 for me. Other than that, thanks!!
